src/renderer/hooks/useInstantOnboarding.ts

```diff
@@ -0,0 +1,196 @@
+import { useCallback, useContext } from 'react'
+
+import useConfirmationDialog from './useConfirmationDialog'
+import useDialog from '../hooks/useDialog'
+import useSecureJoin from './useSecureJoin'
+import useTranslationFunction from './useTranslationFunction'
+import { BackendRemote } from '../backend-com'
+import { ConfigureProgressDialog } from '../components/LoginForm'
+import { InstantOnboardingContext } from '../contexts/InstantOnboardingContext'
+import { ScreenContext } from '../contexts/ScreenContext'
+import { Screens } from '../ScreenController'
+
+import type { QrWithUrl } from './useProcessQr'
+
+const DEFAULT_CHATMAIL_INSTANCE_URL =
+  'https://nine.testrun.org/cgi-bin/newemail.py'
+
+export default function useInstantOnboarding() {
+  const context = useContext(InstantOnboardingContext)
+  const openConfirmationDialog = useConfirmationDialog()
+  const tx = useTranslationFunction()
+  const { openDialog } = useDialog()
+  const { screen, changeScreen } = useContext(ScreenContext)
+  const { secureJoinContact, secureJoinGroup } = useSecureJoin()
+
+  if (!context) {
+    throw new Error(
+      'useInstantOnboarding has to be used within <InstantOnboardingProvider>'
+    )
+  }
+
+  const {
+    setShowInstantOnboarding,
+    setWelcomeQr,
+    showInstantOnboarding,
+    welcomeQr,
+  } = context
+
+  const switchToInstantOnboarding = useCallback(
+    async (qrWithUrl?: QrWithUrl) => {
+      if (qrWithUrl) {
+        const { qr } = qrWithUrl
+        if (
+          !['account', 'askVerifyGroup', 'askVerifyContact'].includes(qr.kind)
+        ) {
+          throw new Error(
+            'QR code needs to be of kind `account`, `askVerifyGroup` or `askVerifyContact` for instant onboarding flow'
+          )
+        }
+
+        // Ask user to confirm creating a new account if they already have one
+        const numberOfAccounts = (await BackendRemote.rpc.getAllAccountIds())
+          .length
```

Comment:
you could also ask if the current account is unconfigured, if it is use it, if it is configured ask to add a new account

Reply:
I guess if the account is unconfigured you do not even need to ask the user here.